src/bp/core/services/workspace-service.ts

```diff
@@ -11,115 +11,117 @@ import { GhostService } from './ghost/service'
+    await this.save(workspace)
   }
 
   async deleteBotRef(botId: any): Promise<void> {
-    const index = this._workspace.bots.findIndex(x => x === botId)
-    this._workspace.bots.splice(index, 1)
-    await this.save()
+    const workspace = await this.getWorkspace()
+    const index = workspace.bots.findIndex(x => x === botId)
+    workspace.bots.splice(index, 1)
+    await this.save(workspace)
   }
 
-  getBotRefs(): string[] {
-    return this._workspace.bots
+  async getBotRefs(): Promise<string[]> {
+    const workspace = await this.getWorkspace()
+    return workspace.bots
   }
 
-  listUsers(selectFields?: Array<keyof AuthUser>): Partial<AuthUser[]> {
+  async listUsers(selectFields?: Array<keyof AuthUser>): Promise<Partial<AuthUser[]>> {
+    const workspace = await this.getWorkspace()
     if (!selectFields) {
-      return this._workspace && this._workspace.users
+      return workspace.users
     } else {
-      return _.map(this._workspace.users, x => _.pick(x, selectFields))
+      return _.map(workspace.users, x => _.pick(x, selectFields))
     }
   }
 
-  assertUserExists(email: string): void {
-    const user = this.findUser({ email })
+  async assertUserExists(email: string): Promise<void> {
+    const user = await this.findUser({ email })
     if (!user) {
-      throw new Error(`User "${email}" is not a part of the workspace "${this._workspace.name}"`)
+      throw new Error(`User "${email}" is not a part of the workspace`)
     }
   }
 
   async assertIsRootAdmin(role: string) {
-    if (this._workspace.roles.find(r => r.id === 'admin') && role !== 'admin') {
+    const workspace = await this.getWorkspace()
+    if (workspace.roles.find(r => r.id === 'admin') && role !== 'admin') {
       throw new UnauthorizedAccessError(`Only root admin is allowed to use this`)
     }
   }
 
-  findUser(where: {}, selectFields?: Array<keyof AuthUser>): AuthUser | undefined {
-    const user = _.head(_.filter(this._workspace.users, where))
+  async findUser(where: {}, selectFields?: Array<keyof AuthUser>): Promise<AuthUser | undefined> {
+    const workspace = await this.getWorkspace()
+    const user = _.head(_.filter(workspace.users, where))
     // return selectFields ? _.pick(user, selectFields) : user
     return user
   }
 
-  findRole(roleId: string): AuthRole {
-    const role = this._workspace.roles.find(r => r.id === roleId)
+  async findRole(roleId: string): Promise<AuthRole> {
+    const workspace = await this.getWorkspace()
+    const role = workspace.roles.find(r => r.id === roleId)
     if (!role) {
-      throw new Error(`Role "${roleId}" does not exists in workspace "${this._workspace.name}"`)
+      throw new Error(`Role "${roleId}" does not exists in workspace "${workspace.name}"`)
     }
     return role
   }
 
-  getRoleForUser(email: string): AuthRole {
-    const user = this.findUser({ email })!
-    return this.findRole(user.role!)
+  async getRoleForUser(email: string): Promise<AuthRole | undefined> {
```

Comment:
`undefined` this cant happen because we throw in `findRole` if role is undefined